src/wifi.rs

```diff
@@ -545,17 +545,22 @@ impl EspWifi {
 
         self.shared.with_lock(|shared| shared.operating = false);
 
-        esp!(unsafe { esp_wifi_disconnect() }).or_else(|err| {
-            if err.code() == esp_idf_sys::ESP_ERR_WIFI_NOT_STARTED as esp_err_t {
-                Ok(())
-            } else {
-                Err(err)
-            }
-        })?;
-        info!("Disconnect requested");
+        let status = self.get_status();
```

Comment:
This protection is not necessary, as the code below is protected already?

Reply:
I'll try to check again